src/dowel/csv_output.py

```diff
@@ -42,20 +42,42 @@ def record(self, data, prefix=''):
                 self._writer.writeheader()
 
             if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
-
-            self._writer.writerow(to_csv)
+                self._fieldnames |= to_csv.keys()
+                if set(to_csv.keys()).issubset(self._fieldnames):
```

Comment:
It's a little unclear to me what the logic means here. Does this only apply if a field is removed?

Reply:
This adds all the keys from the new to_csv dictionary into _fieldnames